test/integration/targets/concurrency/runme.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+
+set -eux -o pipefail
+
+# set `fileno` limit artificially low to induce "too many open files" failures with a low number of forks
+ulimit -n 30
+
+echo "running test playbook and capturing output..."
+
+# run test playbook with enough hosts/forks to quickly exhaust the low open files limit
+set +e
+ansible-playbook -i hosts -f 20 test.yml 2>&1 | tee out.txt
+RC=$?
+set -e
+
+echo "ansible-playbook should yield RC==250 (unexpected error)"
+[[ $RC == 250 ]] && echo PASS
```

Comment:
should the test give an error if these fail?
```suggestion
[[ $RC == 250 ]] && echo PASS || { echo "Error: RC is not equal to 250"; exit 1; }
```

Reply:
They could, but the combination of the pre-statement `echo`s and the `set +x` output should cover both bases. Every time I have to do it, I play with different approaches to `runme.sh` testing, and I still haven't found anything I really like yet :laughing:

Reply:
There's probably a more succinct way to do it than my suggestion, I always have to fiddle with these so long. I checked out the test without the fix though, and they are both passing, albeit without echoing PASS.

<details>
<summary>test...</summary>
<br>

```bash
[shertel@fedora ansible]$ git log -n2
commit 0aa14fad4377db69ab964a82530a8f67f42fd6b0 (HEAD -> pr/82480)
Author: Matt Davis <nitzmahone@redhat.com>
Date:   Wed Jan 3 17:31:52 2024 -0800

    propagate WorkerProcess create/start exceptions
    
    * failure to start a worker process for any reason should be immediately fatal
    * add hint text for OSError 23/24 (too many open files)

commit e1bcb8a534a7144da937bc658d5a59f1de24dd0f
Author: Matt Clay <matt@mystile.com>
Date:   Wed Jan 3 14:29:38 2024 -0800

    Remove unused unit test code
[shertel@fedora ansible]$ git diff
[shertel@fedora ansible]$ git diff --staged
[shertel@fedora ansible]$ git checkout e1bcb8a534a7144da937bc658d5a59f1de24dd0f -- lib/ansible/plugins/strategy/
[shertel@fedora ansible]$ git diff --staged --name-only
lib/ansible/plugins/strategy/__init__.py
lib/ansible/plugins/strategy/linear.py
[shertel@fedora ansible]$ ansible-test integration concurrency --docker
Starting new "ansible-test-controller-eS3fcHZS" container.
Adding "ansible-test-controller-eS3fcHZS" to container database.
Installing requirements for Python 3.12 (controller)
Running concurrency integration test script
+ ulimit -n 30
+ echo 'running test playbook and capturing output...'
running test playbook and capturing output...
+ set +e
+ ansible-playbook -i hosts -f 20 test.yml
+ tee out.txt

PLAY [many] ********************************************************************

TASK [some workers will fail] **************************************************
changed: [host01]
changed: [host02]
changed: [host03]
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 24] Too many open files
fatal: [host04]: FAILED! => {"msg": "Unexpected failure during module execution: [Errno 24] Too many open files", "stdout": ""}
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: [Errno 24] Too many open files
fatal: [host05]: FAILED! => {"msg": "Unexpected failure during module execution: [Errno 24] Too many open files", "stdout": ""}

TASK [should not run] **********************************************************

PLAY RECAP *********************************************************************
host01                     : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
host02                     : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
host03                     : ok=1    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
host04                     : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
host05                     : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

+ RC=2
+ set -e
+ echo 'ansible-playbook should yield RC==250 (unexpected error)'
ansible-playbook should yield RC==250 (unexpected error)
+ [[ 2 == 250 ]]
+ echo ''\''some workers will fail'\'' task ran?'
'some workers will fail' task ran?
+ grep 'some workers will fail' out.txt
TASK [some workers will fail] **************************************************
+ echo PASS
PASS
+ echo ''\''should not run'\'' task did not run?'
'should not run' task did not run?
+ grep 'should not run' out.txt
TASK [should not run] **********************************************************
+ echo 'PASS: all tests ok'
PASS: all tests ok
[shertel@fedora ansible]$ $?
bash: 0: command not found...
```
</details>